ui/StatusQ/src/submodelproxymodel.cpp

```diff
@@ -41,31 +39,34 @@ QVariant SubmodelProxyModel::data(const QModelIndex& index, int role) const
             return submodel;
         }
 
-        QVariant attachement = submodelObj->property(attachementPropertyName);
+        QVariant proxyVariant;
 
-        if (attachement.isValid())
-            return attachement;
+        auto& entry = m_container[index.row()];
 
-        // Make sure that wrapper is destroyed before it receives signal related
-        // to submodel's destruction. Otherwise injected context property may
-        // be cleared causing warnings related to accessing null from qml.
-        connect(submodelObj, &QObject::destroyed, this, [](auto obj) {
-            QVariant attachement = obj->property(attachementPropertyName);
+        if (entry) {
+            proxyVariant = QVariant::fromValue(entry.get());
+        }
 
-            if (attachement.isValid())
-                delete attachement.value<QObject*>();
-        });
+        if (proxyVariant.isValid())
+            return proxyVariant;
 
         auto creationContext = m_delegateModel->creationContext();
         auto parentContext = creationContext
                 ? creationContext : m_delegateModel->engine()->rootContext();
 
         auto context = new QQmlContext(parentContext, submodelObj);
+
+        // Make sure that wrapper is destroyed before it receives signal related
+        // to submodel's destruction. Otherwise injected context property may
+        // be cleared causing warnings related to accessing null from qml.
+        connect(submodelObj, &QObject::destroyed, this,
+                [context = QPointer(context)](auto obj) {
+            delete context.data();
```

Comment:
I'd even do the safer way, since it's a `QObject`, and use `deleteLater()` instead

Reply:
```suggestion
            context->deleteLater();
```

Reply:
Ah I see... so it's already too late :/ Unfortunately I can't think of a better way to catch the destruction earlier